src/Dfc.CourseDirectory.Web/Controllers/ProviderApprenticeships/ProviderApprenticeshipsController.cs

```diff
@@ -67,62 +35,65 @@ public async Task<IActionResult> Index(Guid? apprenticeshipId, string message)
             {
-                var allLiveProviderApprenticeships =
-                    result.Value.Where(x => x.RecordStatus == RecordStatus.Live);
+                Predicate = a =>
+                    a.ProviderUKPRN == UKPRN
+                    && a.RecordStatus == (int)ApprenticeshipStatus.Live
+            });
 
-                model.Apprenticeships = new List<Apprenticeship>();
-                foreach(var apprenticeship in allLiveProviderApprenticeships)
+            if (apprenticeshipId.HasValue)
+            {
+                var linkMessage = $"<a id=\"apprenticeshiplink\" class=\"govuk-link\" href=\"#\" data-apprenticeshipid=\"{apprenticeshipId.Value}\">{message}</a>";
+                
+                if (!string.IsNullOrEmpty(message))
                 {
-                    model.Apprenticeships.Add(apprenticeship);
+                    ViewBag.Message = linkMessage;
                 }
-            }
 
-            if (apprenticeshipId.HasValue)
-            {
-                var linkMessage =
-                    $"<a id=\"apprenticeshiplink\" class=\"govuk-link\" href=\"#\" data-apprenticeshipid=\"{apprenticeshipId.Value}\">{message}</a>";
-                if (!string.IsNullOrEmpty(message)) ViewBag.Message = linkMessage;
                 ViewBag.ApprenticeshipId = apprenticeshipId.Value;
             }
 
-            return View(model);
+            return View(new ProviderApprenticeshipsViewModel
+            {
+                Apprenticeships = liveApprenticeships.Values.Select(Apprenticeship.FromCosmosDbModel).ToList()
+            });
         }
 
-
         [Authorize]
         public async Task<IActionResult> ProviderApprenticeshipsSearch([FromQuery] ProviderApprenticeShipSearchRequestModel requestModel)
         {
-
-            ProviderApprenticeshipsSearchResultModel model = new ProviderApprenticeshipsSearchResultModel();
-            
             int? UKPRN = Session.GetInt32("UKPRN");
-            var result = await _apprenticeshipService.GetApprenticeshipByUKPRN(UKPRN.ToString());
-
 
-            if (result.IsSuccess)
+            var liveApprenticeships = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeships
             {
-                var liveApprenticeships = result.Value.Where(a => a.RecordStatus == RecordStatus.Live);
+                Predicate = a =>
+                    a.ProviderUKPRN == UKPRN
+                    && a.RecordStatus == (int)ApprenticeshipStatus.Live
+            });
 
-                if (string.IsNullOrWhiteSpace(requestModel.SearchTerm))
-                {
-                    model.Items = liveApprenticeships.ToList();
-                }
-                else
-                {
-                    var searchTermWords = requestModel.SearchTerm.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+            var model = new ProviderApprenticeshipsSearchResultModel();
+
+            if (string.IsNullOrWhiteSpace(requestModel.SearchTerm))
```

Comment:
I didn't really want to re-implement search, but it was here... https://github.com/SkillsFundingAgency/dfc-providerportal-apprenticeships/blob/main/src/Dfc.ProviderPortal.Apprenticeships/Helper/CosmosDbHelper.cs#L170-L222